src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -782,19 +782,6 @@ impl<'a, 'tcx> Visitor<'tcx> for Checker<'a, 'tcx> {
 
                 // Only allow statics (not consts) to refer to other statics.
                 if self.mode == Mode::Static || self.mode == Mode::StaticMut {
-                    if self.mode == Mode::Static
```

Comment:
you also need to remove it from https://github.com/rust-lang/rust/blob/56237d75b4271a8a2e0f47d86ea76ebf6d966152/src/librustc_mir/transform/check_consts/validation.rs#L337

Reply:
and there's a test: https://github.com/rust-lang/rust/blob/master/src/test/ui/consts/const-eval/assign-to-static-within-other-static.rs#L10

Reply:
Added the missing bits.